pkg/storage/endpoint/tso.go

```diff
@@ -64,7 +66,33 @@ func (se *StorageEndpoint) LoadTimestamp(prefix string) (time.Time, error) {
 }
 
 // SaveTimestamp saves the timestamp to the storage.
-func (se *StorageEndpoint) SaveTimestamp(key string, ts time.Time) error {
-	data := typeutil.Uint64ToBytes(uint64(ts.UnixNano()))
-	return se.Save(key, string(data))
+func (se *StorageEndpoint) SaveTimestamp(prefix string, key string, ts time.Time) error {
+	return se.RunInTxn(context.Background(), func(txn kv.Txn) error {
+		prefixEnd := clientv3.GetPrefixRangeEnd(prefix)
+		keys, values, err := txn.LoadRange(prefix, prefixEnd, 0)
+		if err != nil {
+			return err
+		}
+
+		previousTS := typeutil.ZeroTime
+		for i, key := range keys {
+			key := strings.TrimSpace(key)
+			if !strings.HasSuffix(key, timestampKey) {
```

Comment:
Isn't this condition "!strings.HasSuffix(key, timestampKey)" too loose, considering we're using "/pd/[cluster-id]" to do prefix range scan? Could there be chance that 'timestamp' suffix are added to the path for other use cases?